### PR TITLE
Implement static field base and offset substitutions for Unsafe

### PR DESCRIPTION
**Pull request by ivan-ristovic**
This PR implements `staticFieldOffset` and `staticFieldBase` methods in `com.oracle.svm.core.jdk.SunMiscUnsafeSubstitutions`

### CODE REVIEW

File: substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/target/Target_jdk_internal_misc_Unsafe_Reflection.java [hunk 1]
```diff
@@ -67,4 +68,9 @@ public long objectFieldOffset(Class<?> c, String name) {
             throw new InternalError();
         }
     }
+
+    @Substitute
+    private long staticFieldOffset(Target_java_lang_reflect_Field f) {
+        return f.offset;
```
Comment (vjovanov) [inline substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/target/Target_jdk_internal_misc_Unsafe_Reflection.java:74]: What happens if we don't have the field in the configuration? Don't we need a similar check as in the object version of the method?

Comment (ivan-ristovic) [reply substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/target/Target_jdk_internal_misc_Unsafe_Reflection.java]: Fixed.

Comment (ivan-ristovic) [reply substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/target/Target_jdk_internal_misc_Unsafe_Reflection.java]: Also added a comment as clarification as to why the implementation is the same.

**Review by peter-hofer — APPROVED** [review]
Thank you @ivan-ristovic !

**Comment by peter-hofer** [comment]
@ivan-ristovic , have you already signed and submitted the Oracle Contributor Agreement (OCA)? If not, please follow the instructions in step II here: https://www.graalvm.org/community/contributors/

**Comment by ivan-ristovic** [comment]
Yes I have, and I have received a response long time ago. I assumed that the bot will automatically pick that up.

**Comment by peter-hofer** [comment]
Looks like @cstancu has already merged a PR with static fields support in Unsafe: 7df688426f38966e40376a2db7cdd93a583ff810. Sorry about the timing, but thank you for your contribution!

Regarding the OCA, I have manually added you to our database so the bot should be satisfied alone in the future.

**Comment by cstancu** [comment]
@ivan-ristovic sorry for this, searching through the issues mentioning static field unsafe support I only found #2694 and #3020, not this PR unfortunately.

**Comment by ivan-ristovic** [comment]
@cstancu No worries!
@peter-hofer Roger, thanks!

**Pull request closed without merging**